source/server/http_time_tracking_filter.cc

```diff
@@ -0,0 +1,68 @@
+#include "server/http_time_tracking_filter.h"
+
+#include <string>
+
+#include "envoy/server/filter_config.h"
+
+#include "server/configuration.h"
+#include "server/well_known_headers.h"
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+
+namespace Nighthawk {
+namespace Server {
+
+HttpTimeTrackingFilterConfig::HttpTimeTrackingFilterConfig(
+    nighthawk::server::ResponseOptions proto_config)
+    : server_config_(std::move(proto_config)) {}
+
+uint64_t
+HttpTimeTrackingFilterConfig::getElapsedNanosSinceLastRequest(Envoy::TimeSource& time_source) {
+  return getRequestStopwatch().getElapsedNsAndReset(time_source);
+}
+
+HttpTimeTrackingFilter::HttpTimeTrackingFilter(HttpTimeTrackingFilterConfigSharedPtr config)
+    : config_(std::move(config)) {}
+
+Envoy::Http::FilterHeadersStatus
+HttpTimeTrackingFilter::decodeHeaders(Envoy::Http::RequestHeaderMap& headers, bool /*end_stream*/) {
```

Comment:
My 2p:

1. I was thinking that this doesn't hurt, and may avoid a future reader having to look it up? 
2. For context: we derive from `PassThroughFilter` it as it conveniently stubs all the overrides precisely like we need, allowing us to override and focus on just the virtual methods of the interface that we are interested in. Maybe this deserves a code-level comment, wdyt?  By throwing an error, do you mean that it replies with a 500 when it doesn't understand the configuration? Could you elaborate on the nature of the potential problem you would like to rule out?

Reply:
1. Sounds good.  I was going back and forth on whether it was helpful.  Decided to raise the question to hear your opinion.
2. I did mean reply with a 500, sorry for being unclear.  I think all of that makes sense.  Just wasn't sure if we were breaking some sort of envoy contract.  I had looked up PassThroughFilter and it specifies that it always returns Continue.  I wanted to make sure that we were using this as a convenience and it was okay to break that rule, rather than it being a contract for the class that is relied upon